### DISCUSSION TITLE
MidiOut

### DISCUSSION THREAD

**Discussion by pissedjean** [playbenny/benny#484] [Q&A] [answered]
Hello!

Is it possible to route midi out of benny via iac driver bus for example?

Ahoj

Boris

**Comment by jamesholdenmusic** [comment, marked as answer]
Hi Boris, yes it is.

The way I'd envisioned midi in and out of benny is you should set it up in the 'hardware editor' bit. You could add a new hardware device to your config called 'IAC midi' for example, assign the right midi in and out ports to it in the hardware editor, save the profile and then it'll appear as a block in the new block menu. Maybe it makes more sense to have separate IAC in and out blocks? up to you.

The hardware editor is a bit less elegant than the rest of it.. if you hit problems come back and ask and we can work it out.

also when you get it working it'd be cool if you could share your hardware profile file here and i'll add it to the repo (i hadn't thought of that use case, but i guess other people might want to do the same. i don't have a mac myself to do it on)

**Comment by pissedjean** [comment]
Hello James! 

Thanks for getting back. I had a look at the hardware editor but i am still a beginner with Max so it seems a little bit complicated.
I managed to add a new Block but how do i setup the midi in and out ports?
This must be done in the patcheditor right?

Thanks

Ahoj

Boris

**Comment by jamesholdenmusic** [comment]
Hi Boris, thanks for trying. 

So when you add a new block in the hardware editor there should be a button marked "add midi input channel", and when you press it it should add a section like the one pictured below. this section is for midi from benny to the input of IAC midi.

in the dropdown i've highlighted you should be able to select IAC midi? if not, in any max window go options | midi options and check IAC midi is listed and enabled there?

![image](https://github.com/user-attachments/assets/bd5d85e4-fbbb-4899-9180-2d0069515a7a)

If you need to receive midi from another app into benny via IAC then make sure it's selected in the keyboards section (I haven't yet implemented it in a hw block)
Thanks for asking this is useful

**Reply by pissedjean** [reply]
Hello James!

Thanks for getting back and the description.
I used my Torso T1 Sequencer via the input keyboard.
This works fine.
But when i set up a new Hardware Block it's possible to choose the right midi input channel but unfortunately it won't save.
For example if i use no_hardware.json in the editor and add another hardware.newblock with midiin iac driver bus and save it, it will jump in the presetmenu on the first preset and what i choose for midiin will not be saved.
I also tried to save it under a different name but that won't work either.
I did the changes like described above in the no hardware preset and saved it under the name midi then i put the midi.json file in the hardware configs folder but this does also not work.

that's what i tried so far.

thanks + ahoj

boris

**Reply by jamesholdenmusic** [reply]
hi boris
thanks for testing more, appreciated. i introduced a bunch of bugs to the midi section of the editor in the recent improvements. i think that's all fixed now - you can save midi input and output for a hardware block, and the dropdowns should have the right lists in. 
(edit - now works without any messing around, have been testing it here seems solid finally but let me know if not!)

**Reply by pissedjean** [reply]
Hi James!

The hardware blocks seem to work i did as you said. 
Saving worked well and i was able to send midi to external bespokesynth.

So thanks alot for this.

**Reply by jamesholdenmusic** [reply]
great! thanks for helping fix this bit
